### PR TITLE
Use XmlHttpRequest when making requests from a renderer

### PR DESCRIPTION
**Pull request by jrieken**
This is for #16065 and avoids using node API for making http(s) requests

### CODE REVIEW

**Comment by jrieken** [comment]
@joaomoreno The following headers cannot be set when using XHR and I wonder what they are used for: `User-Agent`, `Accept-Endinding`, `Content-Length`

**Comment by joaomoreno** [comment]
The Marketplace guys require that we set the `User-Agent` that we current send. This is a blocker.

`Content-Length` isn't needed when doing XHR. `Accept-Encoding` is used to benefit from the recent Marketplace improvements, in which they cache all responses in gzip format. AFAIK it's up to the browser to add this header with whatever encodings it supports, so that sucks too.

**Comment by jrieken** [comment]
re https://github.com/Microsoft/vscode/pull/16211#issuecomment-263623918 `User-Agent` is set by electron and matches our expectation I'd say, it happens to be *VSCode 1.8.0*, otherwise we have to convince MP folks to accept `X-User-Agent`. Browser happily accept gzip and much more, no issues there.

screen shot 2016-11-29 at 17 43 14

**Comment by jrieken** [comment]
@joaomoreno I have added logic to ignore XHR when a proxy in configured, unsure if I have caught all weird config cases...

**Comment by joaomoreno** [comment]
Oh. `VSCode 1.8.0` is exactly what that value should be. 👍

File: src/vs/platform/request/electron-browser/requestService.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * This service exposes the `request` API, while using the global
+ * or configured proxy settings.
+ */
+export class RequestService extends NodeRequestService {
+	request(options: IRequestOptions): TPromise<IRequestContext> {
+		return super.request(options, xhrRequest);
+	}
+}
+
+class ArryBufferStream extends Readable {
```
Comment (bondz) [inline src/vs/platform/request/electron-browser/requestService.ts]: nit: Don't you mean ArrAyBufferStream?

Comment (jrieken) [inline src/vs/platform/request/electron-browser/requestService.ts]: yeah. typos...

Comment (jrieken) [inline src/vs/platform/request/electron-browser/requestService.ts]: Pirate-Programming

**Pull request merged**